core/domain/collection_services.py

```diff
@@ -1020,14 +1020,57 @@ def compute_summary_of_collection(collection, contributor_id_to_add):
         collection.objective, collection.language_code, collection.tags,
         collection_rights.status, collection_rights.community_owned,
         collection_rights.owner_ids, collection_rights.editor_ids,
-        collection_rights.viewer_ids, contributor_ids, contributors_summary,
+        collection_rights.viewer_ids, collection_summary_model.contributor_ids,
+        collection_summary_model.contributors_summary,
         collection.version, collection_model_node_count,
         collection_model_created_on, collection_model_last_updated
     )
 
     return collection_summary
 
 
+def add_new_contributor_to_collection_summary(
+        collection_summary, contributor_id):
+    """Add a new contributor to the contributors summary in collection summary.
+
+    Args:
+        collection_summary: CollectionSummary. The collection summary.
+        contributor_id: str. ID of the contributor to be added to the collection
+            summary.
+
+    Returns:
+        CollectionSummary. The collection summary with new contributor added to
+            the contributors summary.
+    """
+    contributors_summary = (
+        collection_summary.contributors_summary if collection_summary else {})
+    if contributor_id not in constants.SYSTEM_USER_IDS:
+        contributors_summary[contributor_id] = (
+            contributors_summary.get(contributor_id, 0) + 1)
+
+    collection_summary.contributors_summary = contributors_summary
+    collection_summary.contributor_ids = list(contributors_summary.keys())
+
+    return collection_summary
+
+
+def regenerate_contributors_summary(collection_summary):
+    """Regenerate the contributors summary from the snapshots.
+
+    Args:
+        collection_summary: CollectionSummary. The collection summary.
+
+    Returns:
+        CollectionSummary. The collection summary with regenerated
+            contributors summary.
+    """
+    collection_summary.contributors_summary = (
+        compute_collection_contributors_summary(collection_summary.id))
+    collection_summary.contributor_ids = list(
```

Comment:
This function is also changing the contributors_id, I think this can be done inline around line 995, wdyt? (To avoid confusion with function names and as this method is not used anywhere else and it's very small.)

Reply:
Moved this to `save_collection_summary` where it is even better placed I think.